relayer/relays/beefy/beefy-ethereum-listener.go

```diff
@@ -289,67 +374,55 @@ func (li *BeefyEthereumListener) queryFinalVerificationSuccessfulEvents(ctx cont
-	blockNumber, latestBlockNumber uint64) {
-	// Query previous FinalVerificationSuccessful events and update the status of BEEFY justifications in database
-	events, err := li.queryFinalVerificationSuccessfulEvents(ctx, blockNumber, &latestBlockNumber)
-	if err != nil {
-		log.WithError(err).Error("Failure fetching event logs")
-	}
-	log.Info(fmt.Sprintf(
-		"Found %d FinalVerificationSuccessful events between blocks %d-%d",
-		len(events), blockNumber, latestBlockNumber),
-	)
-
-	for _, event := range events {
-		item := li.beefyDB.GetItemByID(event.Id.Int64())
-		if int64(item.ID) == event.Id.Int64() {
-			log.Infof(
-				"Deleting finalized item %s from the database",
-				event.Id,
-			)
-			deleteCmd := store.NewDatabaseCmd(item, store.Delete, nil)
-			li.dbMessages <- deleteCmd
-		} else {
-			log.Error("BEEFY justification data not found in database for FinalVerificationSuccessful event. Ignoring event.")
-		}
-	}
-}
-
 // processFinalVerificationSuccessfulEvents removes finalized commitments from the relayer's BEEFY justification database
-func (li *BeefyEthereumListener) processFinalVerificationSuccessfulEvents(ctx context.Context,
-	blockNumber uint64) {
-	events, err := li.queryFinalVerificationSuccessfulEvents(ctx, blockNumber, &blockNumber)
+func (li *BeefyEthereumListener) processFinalVerificationSuccessfulEvents(
+	ctx context.Context,
+	startBlock uint64,
+	endBlock uint64,
+) error {
+	events, err := li.queryFinalVerificationSuccessfulEvents(ctx, startBlock, &endBlock)
 	if err != nil {
-		log.WithError(err).Error("Failure fetching event logs")
+		log.WithError(err).Error("Failure querying FinalVerificationSuccessful events")
+		return err
 	}
 
-	if len(events) > 0 {
-		log.Info(fmt.Sprintf("Found %d FinalVerificationSuccessful events on block %d", len(events), blockNumber))
-	}
+	log.WithFields(log.Fields{
+		"startBlock": startBlock,
+		"endBlock":   endBlock,
+		"count":      len(events),
+	}).Debug("Queried for FinalVerificationSuccessful events")
 
 	for _, event := range events {
 		log.WithFields(logrus.Fields{
 			"blockHash":   event.Raw.BlockHash.Hex(),
 			"blockNumber": event.Raw.BlockNumber,
 			"txHash":      event.Raw.TxHash.Hex(),
-		}).Info("event information")
+			"ID":          event.Id.Int64(),
+			"Prover":      event.Prover.Hex(),
+		}).Info("Processing FinalVerificationSuccessful event")
 
 		if event.Prover != li.ethereumConn.GetKP().CommonAddress() {
-			continue
+			log.WithFields(logrus.Fields{
+				"ID":     event.Id.Int64(),
+				"Prover": event.Prover.Hex(),
+			}).Info("Skipping FinalVerificationSuccessful event as it has an unknown prover address")
+			return nil
 		}
 
-		log.Infof(
-			"6: Deleting finalized item %s from the database",
-			event.Id,
-		)
+		item, err := li.beefyDB.GetItemByID(event.Id.Int64())
+		if err != nil {
+			log.Error("Failed to retrieve item from Beefy DB")
+			return err
```

Comment:
again, not sure if we want to actually return error, or just ignore this?

Reply:
I think as a general principle we want the relay to fail-fast on any unexpected error, 1) to be able to reset things to a known good state, 2) to make the error visible so we can address any flaws.

The fetches from the underlying SQL db could fail for many reasons, including IO errors, or a bug in a previous step which needed to initialize the item in the DB. If we don't handle the error, the underlying error could manifest in other problems down the line.

Reply:
I've updated the error handling to handle both yours and mine's concerns. The new code will just skip if no record is found, and fail fast on any other error.